Hoopy's first draft of the install docs

matthewrevell opened the pull request:
This commit reworks the Nakama developer install docs, including:

 * Redrafted intro page.
 * Separated the types of install into different guides.
 * Recommend Docker-based install.

Work still to be done includes upgrades and instructions on specifying local storage for the containers.

mofirouz commented on docs/index.adoc:
```diff
@@ -1,48 +1,65 @@
+  Authenticate player logins using Facebook, Google, Steam and other providers.
 * image:/images/svg/data-backup.svg[] link:./development/storage.adoc[*Data Storage*]
-  Create records for each user and global ones shared for all users. This could be save games, maps, items, and other gameplay objects.
+  Store and retrieve data – such as game saves, maps, highscore board, inventory items and other gameplay objects – using Nakama as your distrbuted data store.
```
"distrbuted" -> "distributed"

matthewrevell commented:
I've fixed the typo, thanks Mo.

I've now also reworked the Docker upgrade path; let me know if you're happy with this.

I think this is ready to merge but please do give it a sanity check. @ukmadlz please check this too.

novabyte commented:
@matthewrevell Nice 👍 I'll wait for @ukmadlz to review final edits and drop a note here before I merge it onto master.

ukmadlz commented:
LGTM

matthewrevell commented:
Mike and I discussed the upgrades for Docker and what I wrote isn't ideal. Let me push a revision before you merge.

matthewrevell commented:
Okay, I've now added Mike's updates to the XXXs and fixed the incorrect Docker upgrade instructions. I'm happy for this to merge but we might go back and make minor changes to some of this section and then bundle them with other PRs.

novabyte commented:
@matthewrevell Perfect. I'll merge it 👍
